### PR TITLE
Fix for md parser ignoring newlines

### LINKED ISSUE #392: [BUG] MD Parser Issue with new lines
new lines are being stripped out when submitting posts using only MD formatting

### PR DESCRIPTION
**Pull request by akinsey**
fixes #392

Test that new lines work in the post body

### CODE REVIEW

File: modules/ept-md-parser/client/marked.min.js [hunk 1]
```diff
@@ -3,4 +3,4 @@
...
...de===token.text){return--pending||done()}token.text=code;token.escaped=true;--pending||done()})})(tokens[i])}return}try{if(opt)opt=merge({},marked.defaults,opt);return Parser.parse(Lexer.lex(src,opt),opt)}catch(e){e.message+="\nPlease report this to https://github.com/chjj/marked.";if((opt||marked.defaults).silent){return"<p>An error occured:</p><pre>"+escape(e.message+"",true)+"</pre>"}throw e}}marked.options=marked.setOptions=function(opt){merge(marked.defaults,opt);return marked};marked.defaults={gfm:true,tables:true,breaks:false,pedantic:false,sanitize:false,sanitizer:null,mangle:true,smartLists:false,silent:false,highlight:null,langPrefix:"lang-",smartypants:false,headerPrefix:"",renderer:new Renderer,xhtml:false};marked.Parser=Parser;marked.parser=Parser.parse;marked.Renderer=Renderer;marked.Lexer=Lexer;marked.lexer=Lexer.lex;marked.InlineLexer=InlineLexer;marked.inlineLexer=InlineLexer.output;marked.parse=marked;if(typeof module!=="undefined"&&typeof exports==="object"){module.exports=marked}else if(typeof define==="function"&&define.amd){define(function(){return marked})}else{this.marked=marked}}).call(function(){return this||(typeof window!=="undefined"?window:global)}());
```
Comment (unenglishable) [inline modules/ept-md-parser/client/marked.min.js]: if you have a file where you add a newline at the end, include that change in a separate commit.   it's not clear what this change does unless you specify what it is doing.

File: modules/ept-md-parser/client/parseFactory.js [hunk 2]
```diff
@@ -3,13 +3,18 @@ var renderer;
 
 function parse(input) {
   if (!input) { input = ''; }
-
```
Comment (unenglishable) [inline modules/ept-md-parser/client/parseFactory.js]: include arbitrary newline changes in a separate commit.  review this and either specify that you intended this change, or undo it.

File: modules/ept-md-parser/client/parseFactory.js [hunk 3]
```diff
@@ -3,13 +3,18 @@ var renderer;
...
   // escape double forward slashes before parsing markdown
   input = input.replace(/\/\//g, '&#47;&#47;');
+
+  // manually handle newlines
+  input = input.replace(/\n/g, String.fromCharCode(26));
+
   // compile markdown
   input = compiler(input, { renderer: renderer });
+
```
Comment (unenglishable) [inline modules/ept-md-parser/client/parseFactory.js]: include arbitrary newline changes in a separate commit. review this and either specify that you intended this change, or undo it.

File: modules/ept-md-parser/client/parseFactory.js [hunk 4]
```diff
@@ -3,13 +3,18 @@ var renderer;
...
   // preserve whitespacing
   input = input.replace(/\r\n|\r|\n/g, String.fromCharCode(26));
+
```
Comment (unenglishable) [inline modules/ept-md-parser/client/parseFactory.js]: include arbitrary newline changes in a separate commit. review this and either specify that you intended this change, or undo it.

Comment (akinsey) [reply modules/ept-md-parser/client/parseFactory.js]: These are all intended

File: modules/ept-md-parser/client/parseFactory.js (hunk 3 above)
Comment (akinsey) [reply modules/ept-md-parser/client/parseFactory.js]: These are all intended

File: modules/ept-md-parser/client/parseFactory.js (hunk 2 above)
Comment (akinsey) [reply modules/ept-md-parser/client/parseFactory.js]: These are all intended

File: modules/ept-md-parser/client/marked.min.js (hunk 1 above)
Comment (unenglishable) [reply modules/ept-md-parser/client/marked.min.js]: this is good for now...

**Pull request merged**